scripts/shared/cloud-prepare.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -em -o pipefail
+
+source "${SCRIPTS_DIR}/lib/debug_functions"
+source "${SCRIPTS_DIR}/lib/utils"
+
+readonly GATEWAY_LABEL='submariner.io/gateway=true'
+
+### Functions ###
+
+function cloud_prepare() {
+    [[ ${cluster_subm[$cluster]} = "true" ]] || return 0
+    ! ensure_gateway_exists || return 0
+
+    case "${PROVIDER}" in
+    kind|aws-ocp)
+        "prepare_${PROVIDER//-/_}"
+        ;;
+    *)
+        echo "Unknown PROVIDER ${PROVIDER@Q}."
+        return 1
+    esac
+}
+
+function ensure_gateway_exists() {
```

Comment:
```suggestion
function check_gateway_exists() {
```

since we don’t do anything to _ensure_ it exists (unlike other parts of the PR).